var/spack/repos/builtin/packages/cbtf/package.py

```diff
@@ -48,12 +47,9 @@ class Cbtf(CMakePackage):
     # for rpc
     depends_on("libtirpc", type="link")
 
-    depends_on("boost@1.70.0:")
-
-    # TODO: replace this with an explicit list of components of Boost,
-    # for instance depends_on('boost +filesystem')
-    # See https://github.com/spack/spack/pull/22303 for reference
-    depends_on(Boost.with_default_variants)
+    depends_on(
+        "boost@1.70.0:+date_time+filesystem+preprocessor+signals2+spirit_classic+test+thread"
```

Comment:
The open-ended version range for boost does not work, it faild with boost-1.86.0.

If I remember correctly, path.is_complete() is version-specific:
```py
5 errors found in build log:
     111    In file included from opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/boost-1.86.0-vzz4cvt6zj7dcdodo4usj2ijz
            33hf2dl/include/boost/bind.hpp:30,
     112                     from .stage/spack-stage-cbtf-1.9.4.1-kdbudmyeo2wcwcqfmlpgl3ggsngj46eu/spack-src/libcbtf/Re
            solvePath.cpp:21:
     113    opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/boost-1.86.0-vzz4cvt6zj7dcdodo4usj2ijz33hf2dl/include/boost/
            bind.hpp:36:1: note: '#pragma message: The practice of declaring the Bind placeholders (_1, _2, ...) in the global namespace is d
            eprecated. Please use <boost/bind/bind.hpp> + using namespace boost:placeholders, or define BOOST_BIND_GLOBAL_PLACEHOLDERS to re
            tain the current behavior.'
     114       36 | BOOST_PRAGMA_MESSAGE(
     115          | ^~~~~~~~~~~~~~~~~~~~
     116    .stage/spack-stage-cbtf-1.9.4.1-kdbudmyeo2wcwcqfmlpgl3ggsngj46eu/spack-src/libcbtf/ResolvePath.cpp: In func
            tion 'boost:filesystem:path KrellInstitute:CBTF:Impl:resolvePath(const std:vector<boost:filesystem:path>&, const boost:f
            ilesystem:path&)':
  >> 117    .stage/spack-stage-cbtf-1.9.4.1-kdbudmyeo2wcwcqfmlpgl3ggsngj46eu/spack-src/libcbtf/ResolvePath.cpp:130:14:
            error: 'const class boost:filesystem:path' has no member named 'is_complete'
     118      130 |     if (path.is_complete())
     119          |              ^~~~~~~~~~~
     120    .stage/spack-stage-cbtf-1.9.4.1-kdbudmyeo2wcwcqfmlpgl3ggsngj46eu/spack-src/libcbtf/ResolvePath.cpp: In func
            tion 'boost:filesystem:path KrellInstitute:CBTF:Impl:resolvePath(const FileType&, const boost:filesystem:path&)':
  >> 121    .stage/spack-stage-cbtf-1.9.4.1-kdbudmyeo2wcwcqfmlpgl3ggsngj46eu/spack-src/libcbtf/ResolvePath.cpp:158:14:
            error: 'const class boost:filesystem:path' has no member named 'is_complete'
     122      158 |     if (path.is_complete())
     123          |              ^~~~~~~~~~~
  >> 124    make[2]: *** [libcbtf/CMakeFiles/cbtf.dir/build.make:107: libcbtf/CMakeFiles/cbtf.dir/ResolvePath.cpp.o] Error 1
     125    make[2]: Leaving directory '.stage/spack-stage-cbtf-1.9.4.1-kdbudmyeo2wcwcqfmlpgl3ggsngj46eu/spack-src/buil
            d_cbtf'
  >> 126    make[1]: *** [CMakeFiles/Makefile2:266: libcbtf/CMakeFiles/cbtf.dir/all] Error 2
     127    make[1]: Leaving directory '.stage/spack-stage-cbtf-1.9.4.1-kdbudmyeo2wcwcqfmlpgl3ggsngj46eu/spack-src/buil
            d_cbtf'
  >> 128    make: *** [Makefile:139: all] Error 2

See build log for details:
  .stage/spack-stage-cbtf-1.9.4.1-kdbudmyeo2wcwcqfmlpgl3ggsngj46eu/spack-build-out.txt
```

Reply:
>If I remember correctly, path.is_complete() is version-specific:

I'll check on that.

Reply:
`is_complete` was removed in 1.85.0, so I added a version constraint.